localization/yabloc/yabloc_particle_filter/src/prediction/predictor.cpp

```diff
@@ -53,26 +53,54 @@ Predictor::Predictor()
   auto on_twist_cov = std::bind(&Predictor::on_twist_cov, this, _1);
   auto on_particle = std::bind(&Predictor::on_weighted_particles, this, _1);
   auto on_height = [this](std_msgs::msg::Float32 m) -> void { this->ground_height_ = m.data; };
+  auto on_ekf_pose = [this](PoseCovStamped::ConstSharedPtr m) -> void {
+    this->latest_ekf_pose_ptr_ = m;
+  };
 
   initialpose_sub_ = create_subscription<PoseCovStamped>("~/input/initialpose", 1, on_initial);
   particles_sub_ =
     create_subscription<ParticleArray>("~/input/weighted_particles", 10, on_particle);
   height_sub_ = create_subscription<std_msgs::msg::Float32>("~/input/height", 10, on_height);
   twist_cov_sub_ =
     create_subscription<TwistCovStamped>("~/input/twist_with_covariance", 10, on_twist_cov);
+  ekf_pose_sub_ = create_subscription<PoseCovStamped>("~/input/ekf_pose", 10, on_ekf_pose);
 
   // Timer callback
   const double prediction_rate = declare_parameter<double>("prediction_rate");
   auto on_timer = std::bind(&Predictor::on_timer, this);
   timer_ = rclcpp::create_timer(
     this, this->get_clock(), rclcpp::Rate(prediction_rate).period(), std::move(on_timer));
 
+  // Service server
+  using std::placeholders::_2;
+  auto on_service = std::bind(&Predictor::on_service, this, _1, _2);
+  enable_server_ = create_service<SetBool>("~/yabloc_suspend_srv", on_service);
+
   // Optional modules
   if (declare_parameter<bool>("visualize", false)) {
     visualizer_ptr_ = std::make_unique<ParticleVisualizer>(*this);
   }
 }
 
+void Predictor::on_service(
+  SetBool::Request::ConstSharedPtr request, SetBool::Response::SharedPtr response)
+{
+  RCLCPP_INFO_STREAM(get_logger(), "particle filter activatation is set to " << request->data);
```

Comment:
There seems to be a typo.

Reply:
Fixed https://github.com/autowarefoundation/autoware.universe/pull/5846/commits/aae8e8e1c94510c31f2d7dc76d1bb3b5a50c54a2